docs/tutorials/flask.md

```diff
@@ -0,0 +1,580 @@
+VM, the microk8s built-in container registry should already been enabled, and we
+will use that for this tutorial.
+
+```bash
+# Use the skopeo, an container images utility tool, bundled with rockcraft to upload the rock image to microk8s built-in registry (localhost:32000)
+ubuntu@charm-dev-vm:~/sample-flask$ /snap/rockcraft/current/bin/skopeo --insecure-policy copy --dest-tls-verify=false oci-archive:sample-flask_0.1_amd64.rock docker://localhost:32000/sample-flask:main
+Getting image source signatures
+Copying blob 29202e855b20 done
+Copying blob 1ae6062392ff done
+Copying blob 2125f5130a12 done
+Copying blob 2f1020cd27a0 done
+Copying blob 6ad38169ff53 done
+Copying config cd33d09f74 done
+Writing manifest to image destination
+Storing signatures
+```
+
+Now it's the time to build the charm. We need to create a charm directory
+(`mkdir -p charm/`) inside the Flask project directory to hold the charm related
+files.
+
+```bash
+# make the charm directory and change directory into the new charm directory
+ubuntu@charm-dev-vm:~/sample-flask$ mkdir charm
+ubuntu@charm-dev-vm:~/sample-flask$ cd charm
+ubuntu@charm-dev-vm:~/sample-flask/charm$
+```
+
+In the same Multipass VM shell, inside the charm directory, run
+`charmcraft init --profile flask-framework` to initialise the file tree
+structure for the PaaS app charmer Flask charm, and inspect the result.
+
+```bash
+# Initialise the charm tree structure:
+ubuntu@charm-dev-vm:~/sample-flask/charm$ charmcraft init --profile flask-framework
+Charmed operator package file and directory tree initialised.
+
+Now edit the following package files to provide fundamental charm metadata
+and other information:
+
+charmcraft.yaml
+src/charm.py
+README.md
+```
+
+In your local editor, open the `charmcraft.yaml` file and customise its contents
+as below (you only have to edit the title, summary, and description). You can
+leave the commented part for now, we will revisit some of them later:
+
+```yaml
+# This file configures Charmcraft.
+# See https://juju.is/docs/sdk/charmcraft-config for guidance.
+
+# (Required)
+name: sample-flask
+
+# (Required)
+type: charm
+
+# (Required for 'charm' type)
+bases:
+  - build-on:
+    - name: ubuntu
+      channel: "22.04"
+    run-on:
+    - name: ubuntu
+      channel: "22.04"
+
+# (Required)
+summary: The charm for the sample-flask Flask application.
+
+# (Required)
+description: |
+    The charm for the sample-flask Flask application, built with the PaaS app charmer charmcraft extension.
+
+    This charm will help you to manage your Flask application in Kubernetes environment with the Juju ecosystem.
+
+
+# (Required for enabling PaaS app charmer Flask framework support)
+extensions:
+  - flask-framework
+
+# Uncomment the integrations used by your application
+# requires:
+#   mysql:
+#     interface: mysql_client
+#     limit: 1
+#   postgresql:
+#     interface: postgresql_client
+#     limit: 1
```

Comment:
This will have to be adapted everytime the template changes, should we put it there ?

Reply:
yeh thats a good point, I'll change it to just refer to what should be updated

Reply:
removed